### PR TITLE
#151 - Introducing IntegerNumberField

### PR DESCRIPTION
**Pull request by dve**
Connected to #151

Suggestion of  an `IntegerNumberField`

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-text-field-flow/152)

### CODE REVIEW

File: vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+  }
+
+  /**
+   * Returns the current value of the number field. By default, the empty number field will return
+   * {@code null} .
+   *
+   * @return the current value.
+   */
+  @Override
+  public T getValue() {
```
[the remark this answers is not in the document]
Comment (dve) [reply vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java]: Overrides javadoc

File: vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+  }
+
+  /**
+   * Sets the value of this number field. If the new value is not equal to {@code getValue()}, fires
+   * a value change event.
+   *
+   * @param value the new value
+   */
+  @Override
+  public void setValue(T value) {
```
[the remark this answers is not in the document]
Comment (dve) [reply vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java]: Overrides javadoc

**Comment by gsedlacz** [comment]
Such a field would be much appreciated.
Another option I could imagine would be to allow the backend to define wich implementation of Number is supposed to be used.

**Comment by pekam** [comment]
We have implemented a new web component to accept only integer-input in the client-side: https://github.com/vaadin/vaadin-text-field/pull/415
Now we're working on the Java API on top of that one. Although your PR has similarities to what we're working on, I'd prefer to start that component implementation from scratch. So I'm closing this PR in favor of new work on tickets #221, #222, #223.

**Pull request closed without merging**